assembly/src/assembler/span_builder.rs

```diff
@@ -148,9 +148,13 @@ impl SpanBuilder {
             let ops = self.ops.drain(..).collect();
             let decorators = self.decorators.drain(..).collect();
             target.push(CodeBlock::new_span_with_decorators(ops, decorators));
-        } else if !self.decorators.is_empty() {
-            // this is a bug in the assembler. we shouldn't have decorators added without their
-            // associated operations
+        } else if self
+            .decorators
+            .iter()
+            .any(|(_, decorator)| matches!(decorator, Decorator::AsmOp(_)))
```

Comment:
Should this be `.any()` or `.all()`? Seems like we want to panic only if all decorators are `AsmOp`, right?

Reply:
My understanding is that `AsmOp` are only associated with operations and not with advice injectors so if there are no operations in a span block, there should be no `AsmOp` either. Is that not the case?

Reply:
hmmm - I think you are correct, but then why do we need to check for `AsmOp` decorators. If there are no operations, there shouldn't be any of them either - right?

Reply:
I'm not sure. I though this was an additional check to make sure the assembler worked correctly (in which case maybe we could replace this with a debug_assert). Or we could remove it if this is redundant now.

Reply:
You are right, but I thought maybe this was meant to be an additional check that the assembler worked correctly (in which case we could also replace it with a debug_assert). I'm not sure if this is completely redundant now. Maybe @vlopes11 could confirm or I could look into it more.